server/routes/validations/users.validations.js

```diff
@@ -42,16 +42,40 @@ const updateProfileValidationRules = () => {
 
 const createAgencyValidationRules = () => {
   return [
-    body('agencyName').notEmpty().isString().withMessage('Agency name is required!'),
+    body('agencyName')
+      .notEmpty()
+      .isString()
+      .withMessage('Agency name is required!'),
     body('agencyWebsite').optional(),
+    body('agencyAddress.address1')
+      .notEmpty()
+      .isLength({min: 5, max: undefined})
+      .withMessage('Address must containt at least 5 characters'),
+    body('agencyAddress.address2').optional(),
+    body('agencyAddress.city')
+      .notEmpty()
+      .isLength({min: 2, max: undefined})
+      .withMessage('City must containt at least 2 characters'),
+    body('agencyAddress.state')
+      .notEmpty()
+      .isLength({min: 2, max: undefined})
+      .withMessage('State must containt at least 2 characters'),
+    body('agencyAddress.country')
+      .notEmpty()
+      .isLength({min: 2, max: undefined})
+      .withMessage('Country must containt at least 2 characters'),
+    body('agencyAddress.zipcode')
+      .notEmpty()
+      .isLength({min: 5, max: undefined})
+      .withMessage('Zipcode must containt at least 5 characters'),
     body('agencyPhone')
       .notEmpty()
-      .isNumeric()
       .isLength({ min: 7, max: undefined })
       .withMessage('Phone number must be at least 7 characters long')
-      .matches(/^(?=\d).{7,}$/)
-      .withMessage('Phone number must contain only numbers'),
+      .matches(/^\(?([0-9]{3})\)?([ .-]?)([0-9]{3})\2([0-9]{4})/) 
+      .withMessage('Phone number must match format XXX-XXX-XXXX'),
```

Comment:
this might cause issues, no one writes their phone number like this, better only check if phonenumber is an integer for now

Reply:
Stacy specifically asked for this:
https://donategifts.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=DON&modal=detail&selectedIssue=DON-63

Reply:
So I've just noticed that people type phone numbers in several different ways. 

- 777-777-7777
- 7777777777
- (+1) 777 777 7777
- (777) 777-7777
- 777.777.7777

I mean this can go on, but you get the point though. 
So how do we validate it without making people type again and again to fit our format? 
Suggestions please

Reply:
I would say we go with only numbers and check the phone number on every key down event, or we filter out all non numbers on submit